Gems/AWSClientAuth/Code/Source/AWSClientAuthEditorSystemComponent.cpp

```diff
@@ -0,0 +1,161 @@
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AWSClientAuthEditorSystemComponent.h>
+
+#include <ActionManager/Action/ActionManagerInterface.h>
+
+#include <AWSCoreBus.h>
+
+namespace AWSClientAuth
+{
+    void AWSClientAuthEditorSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AWSClientAuthSystemComponent::Reflect(context);
+
+        AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serialize)
+        {
+            serialize->Class<AWSClientAuthEditorSystemComponent, AWSClientAuthSystemComponent>()
+                ;
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<AWSClientAuthSystemComponent>("AWSClientAuthEditor", "Provides Client Authentication and Authorization implementations")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("System"))
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true);
+            }
+        }
+    }
+
+    void AWSClientAuthEditorSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("AWSClientAuthEditorService"));
+    }
+
+    void AWSClientAuthEditorSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("AWSClientAuthEditorService"));
+    }
+
+    void AWSClientAuthEditorSystemComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("AWSCoreService"));
+    }
+
+    void AWSClientAuthEditorSystemComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType&)
+    {
+    }
+
+    void AWSClientAuthEditorSystemComponent::Init()
+    {
+        AWSClientAuthSystemComponent::Init();
+    }
+
+    void AWSClientAuthEditorSystemComponent::Activate()
+    {
+        AWSClientAuthSystemComponent::Activate();
+
+        AzToolsFramework::ActionManagerRegistrationNotificationBus::Handler::BusConnect();
+    }
+
+    void AWSClientAuthEditorSystemComponent::Deactivate()
+    {
+        AWSClientAuthSystemComponent::Deactivate();
+
+        AzToolsFramework::ActionManagerRegistrationNotificationBus::Handler::BusDisconnect();
+    }
+
+    void AWSClientAuthEditorSystemComponent::OnMenuBindingHook()
+    {
+        constexpr const char* AWSClientAuth[] =
+        {
+             "Client Auth Gem" ,
+             "client_auth_gem" ,
+             ":/Notifications/download.svg",
+             ""
+        };
+
+        AWSCore::AWSCoreEditorRequestBus::Broadcast(&AWSCore::AWSCoreEditorRequests::CreateSubMenu, AWSCore::AWSMenuIdentifier, AWSClientAuth, 100);
```

Comment:
nit: provide const exprs for magic numbers to provide context for future maintainers ie why this 100, how was it set etc?